api/discovery/v1/foreigncluster_methods.go

```diff
@@ -84,3 +87,32 @@ func (fc *ForeignCluster) LoadForeignCA(localClient kubernetes.Interface, localN
 	}
 	return nil
 }
+
+func (fc *ForeignCluster) SetAdvertisement(adv *protocolv1.Advertisement, discoveryClient *crdClient.CRDClient) error {
+	if fc.Status.Advertisement == nil {
+		// Advertisement has not been set in ForeignCluster yet
+		fc.Status.Advertisement = &v1.ObjectReference{
+			Kind:       "Advertisement",
+			Name:       adv.Name,
+			UID:        adv.UID,
+			APIVersion: "protocol.liqo.io/v1",
+		}
+		_, err := discoveryClient.Resource("foreignclusters").Update(fc.Name, fc, metav1.UpdateOptions{})
```

Comment:
I'm not sure, but since you are modifyng the resource status I think here you need to use `UpdateStatus()` method

Reply:
Yes, I think that it would be slightly more efficient